netbox/utilities/request.py

```diff
@@ -17,10 +17,15 @@ def get_client_ip(request, additional_headers=()):
     )
     for header in HTTP_HEADERS:
         if header in request.META:
-            client_ip = request.META[header].split(',')[0].partition(':')[0]
+            ip = request.META[header].split(',')[0].strip()
+            # Check if the IP address is v6 or v4
+            if ip.count(':') > 1:
+                client_ip = ip
+            else:
+                client_ip = ip.partition(':')[0]
             try:
                 return IPAddress(client_ip)
-            except ValueError:
+            except (AddrFormatError, ValueError):
                 raise ValueError(f"Invalid IP address set for {header}: {client_ip}")
```

Comment:
We should be able to lean on `urllib` here to simplify the logic. We just need to pass `f"//{ip}"` to `urlparse()` and take `hostname`:

```
>>> urlparse('//1.2.3.4').hostname
'1.2.3.4'
>>> urlparse('//1.2.3.4:80').hostname
'1.2.3.4'
>>> urlparse('//[2001:db8::1]').hostname
'2001:db8::1'
>>> urlparse('//[2001:db8::1]:80').hostname
'2001:db8::1'
```

Reply:
I just noticed that this approach doesn't work for IPv6 addresses without brackets, which may appear as `HTTP_X_FORWARDED_FOR` values. It really seems like there should be a simple, canonical solution for this.